### ISSUE TITLE
Video audio not stopped even component unmounts

### ISSUE THREAD

**Issue by stmoy** [microsoft/react-native-windows#7087]
(This issue mirrors the one on the community module repo for tracking purposes: https://github.com/react-native-video/react-native-video/issues/2281)

-=-=-

# Bug
## Platform
* Windows UWP

## Environment info
React native info output:

```shell
 System:
    OS: Windows 10 10.0.19042
    CPU: (8) x64 Intel(R) Core(TM) i5-8250U CPU @ 1.60GHz
    Memory: 9.76 GB / 15.87 GB
  Binaries:
    Node: 14.15.3 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.10 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 7.3.0 - C:\Program Files\nodejs\npm.CMD
    Watchman: Not Found
  SDKs:
    Android SDK:
      API Levels: 28, 29, 30
      Build Tools: 28.0.3, 29.0.2, 30.0.3
      System Images: android-29 | Google APIs Intel x86 Atom
      Android NDK: Not Found
    Windows SDK:
      AllowDevelopmentWithoutDevLicense: Enabled
      AllowAllTrustedApps: Enabled
      Versions: 10.0.18362.0, 10.0.19041.0
  IDEs:
    Android Studio: Version  4.1.0.0 AI-201.8743.12.41.6953283
    Visual Studio: 16.8.30804.86 (Visual Studio Community 2019)
  Languages:
    Java: Not Found
    Python: Not Found
  npmPackages:
    @react-native-community/cli: Not Found
    react: 16.13.1 => 16.13.1
    react-native: 0.63.4 => 0.63.4
    react-native-windows: ^0.63.11 => 0.63.19
  npmGlobalPackages:
    *react-native*: Not Found
```

Library version: x.x.x

## Steps To Reproduce
```
const [videoPlayer, setVideoPlayer] = useState(undefined); 
useEffect(() => {
    return () => {
      setisVisible(false);
      videoPlayer;
    };
  }, []);
 return (
            <Video
              ref={(videoPlayer) => setVideoPlayer(videoPlayer)}/>
           );
```

## Expected behaviour
1. When moving to other screens using react-navigation, video should stop
2. Also the audio for video

## Whats Happening
1. Even when the component unmounts i can hear the audio

**Comment by chrisglein** [comment]
Related to the issues with `display:none` not working (@AgneLukoseviciute)? Or with navigation pages not being disabled ([Gallery#123](https://github.com/microsoft/react-native-gallery/issues/123))?

**Comment by AgneLukoseviciute** [comment]
> Related to the issues with `display:none` not working (@AgneLukoseviciute)? Or with navigation pages not being disabled ([Gallery#123](https://github.com/microsoft/react-native-gallery/issues/123))?

Pasting in a comment from community module repo issue:

>@TomVuillemin is right, we're seeing this cause performance issues in the RNW gallery (Gallery#123) and I believe we are planning on adding windows support for react-native-screens which would likely help here.
>
>There was also a recent change in RNW for fixing XAML Visibility behavior for the display:none property which is used by navigation. If the `<Video />` Component is configured to pause when visibility is set to collapsed, then the display:none fix should be enough to address this issue. It will be included in the upcoming 0.65 release, but we could also backport it to 0.64 if needed.
>
>@jonthysell -- Do you know if the video would pause if collapsed?
>
>In the mean time, a potential workaround would be to use the 'useFocusEffect' hook from navigation along with the 'paused' prop in `<Video />` to make the Video pause when the page is out of focus. For an example, check App.tsx in https://github.com/AgneLukoseviciute/react-navigation-repro on how this workaround was used to make the `<TextInput />` un-editable when leaving a page in react-navigation.

**Comment by chiaramooney** [comment]
@stmoy We should be able to close the copy of this issue present in the react-native-video repo, since this is a Windows navigation issue. Once we get the work wrapped up with react-native-screens, this issue should resolve itself. I'll leave this issue open, so we can make sure to check this case is resolved.
